### PR TITLE
Limit tx set size based on overlay limits

### PR DESCRIPTION
**Pull request by marta-lokhova**

### CODE REVIEW

File: src/herder/test/HerderTests.cpp [hunk 1]
```diff
@@ -1530,6 +1530,110 @@ surgeTest(uint32 protocolVersion, uint32_t nbTxs, uint32_t maxTxSetSize,
...
+
+        while (totalSize < MAX_TX_SET_ALLOWANCE)
+        {
+            auto a = root.create(fmt::format("A{}", txCount++), 500000000);
+            txs.emplace_back(makeTx(a, phase));
+            totalSize += xdr::xdr_size(txs.back()->getEnvelope());
+        }
+
+        TxSetFrame::TxPhases invalidPhases;
+        invalidPhases.resize(2);
```
[the remark this answers is not in the document]
Comment (marta-lokhova) [reply src/herder/test/HerderTests.cpp:1596]: good call, fixed.

File: src/overlay/Peer.h [hunk 2]
```diff
@@ -22,7 +22,14 @@ namespace stellar
 
 typedef std::shared_ptr<SCPQuorumSet> SCPQuorumSetPtr;
 
-static auto const MAX_MESSAGE_SIZE = 0x1000000;
+static auto const MAX_MESSAGE_SIZE = 0x1000000;    // 16 MB
```
[the remark this answers is not in the document]
Comment (marta-lokhova) [reply src/overlay/Peer.h:25]: sure, updated both.

File: src/ledger/LedgerManagerImpl.cpp [hunk 3]
```diff
@@ -475,7 +475,7 @@ LedgerManagerImpl::maxTransactionResources(bool isSoroban,
     }
     else
     {
-        return Resource(1);
+        return Resource(MAX_OPS_PER_TX);
```
[the remark this answers is not in the document]
Comment (marta-lokhova) [reply src/ledger/LedgerManagerImpl.cpp:478]: done.

**Comment by dmkozh** [comment]
r+ 99db143

**Pull request merged**